fearless/Modules/Staking/Analytics/AnalyticsValidators/ViewModel/AnalyticsValidatorsViewModelFactory.swift

```diff
@@ -0,0 +1,287 @@
+                secondLine: totalRewardsText,
+                thirdLine: "100%",
+                locale: locale
+            )
+        }
+    }
+
+    private func createChartCenterText(
+        firstLine: String,
+        secondLine: String,
+        thirdLine: String,
+        locale _: Locale
+    ) -> NSAttributedString {
+        let paragraphStyle = NSMutableParagraphStyle()
+        paragraphStyle.alignment = NSTextAlignment.center
+
+        let activeStakingText = NSAttributedString(
+            string: firstLine,
+            attributes: [
+                NSAttributedString.Key.foregroundColor: R.color.colorAccent()!,
+                NSAttributedString.Key.font: UIFont.capsTitle,
+                NSAttributedString.Key.paragraphStyle: paragraphStyle
+            ]
+        )
+
+        let percentsText = NSAttributedString(
+            string: secondLine,
+            attributes: [
+                NSAttributedString.Key.foregroundColor: R.color.colorWhite()!,
+                NSAttributedString.Key.font: UIFont.h2Title,
+                NSAttributedString.Key.paragraphStyle: paragraphStyle
+            ]
+        )
+
+        let erasRangeText = NSAttributedString(
+            string: thirdLine,
+            attributes: [
+                NSAttributedString.Key.foregroundColor: R.color.colorLightGray()!,
+                NSAttributedString.Key.font: UIFont.h5Title,
+                NSAttributedString.Key.paragraphStyle: paragraphStyle
+            ]
+        )
+
+        let result = NSMutableAttributedString(attributedString: activeStakingText)
+        result.append(.init(string: "\n"))
+        result.append(percentsText)
+        result.append(.init(string: "\n"))
+        result.append(erasRangeText)
+        return result
+    }
+
+    func totalErasCount(eraValidatorInfos: [SubqueryEraValidatorInfo]) -> Int {
+        let distinctEras = Set<EraIndex>(eraValidatorInfos.map(\.era))
+        return distinctEras.count
+    }
+
+    func totalRewardOfStash(
+        address: AccountAddress,
+        rewards: [SubqueryRewardItemData]
+    ) -> Double {
+        let rewardsOfStash = rewards.filter { $0.stashAddress == address }
+        let totalAmount = rewardsOfStash.reduce(Decimal(0)) { amount, info in
+            let decimal = Decimal.fromSubstrateAmount(
+                info.amount,
+                precision: self.chain.addressType.precision
+            )
+            return amount + (decimal ?? 0.0)
+        }
+        return NSDecimalNumber(decimal: totalAmount).doubleValue
+    }
+
+    private func findInactiveSegmentValue(
+        page: AnalyticsValidatorsPage,
+        eraValidatorInfos: [SubqueryEraValidatorInfo],
+        totalEras: Int
+    ) -> Double? {
+        guard case .activity = page else {
+            return nil
+        }
+        let erasRange: Range<EraIndex> = {
+            let eras = eraValidatorInfos.map(\.era)
+            return (eras.min() ?? 0) ..< (eras.max() ?? 0)
```

Comment:
Why it is a range? I suggest just reuse `distinctEras` suggested for implementation in `eraValidatorInfos`